File: app-android/src/main/java/com/ustadmobile/port/android/view/ScormPackageActivity.java
package com.ustadmobile.port.android.view;

import android.os.Build;
import android.os.Bundle;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.toughra.ustadmobile.R;
import com.ustadmobile.core.controller.ScormPackagePresenter;
import com.ustadmobile.core.view.ScormPackageView;

public class ScormPackageActivity extends ZippedContentActivity implements ScormPackageView {

    private WebView mWebView;

    private ScormPackagePresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scorm_package);

        mWebView = (WebView)findViewById(R.id.activity_scorm_package_webview);
        if(Build.VERSION.SDK_INT >= 17) {
            mWebView.getSettings().setMediaPlaybackRequiresUserGesture(false);
        }

        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.getSettings().setDomStorageEnabled(true);
        mWebView.getSettings().setCacheMode(WebSettings.LOAD_DEFAULT);
        mWebView.setWebViewClient(new WebViewClient());
        mWebView.setWebChromeClient(new WebChromeClient());


        setUMToolbar(R.id.um_toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        mPresenter = new ScormPackagePresenter(this, this);
    }

    @Override
    public void setTitle(String title) {
        super.setTitle(title);
    }


    @Override
    public void onDestroy() {
        mPresenter.onDestroy();
        super.onDestroy();
    }

    @Override
    public void loadUrl(String url) {
        mWebView.loadUrl(url);
    }
}

File: app-android/src/main/kotlin/com/ustadmobile/port/android/view/ScormPackageActivity.kt
package com.ustadmobile.port.android.view

import android.os.Build
import android.os.Bundle
import android.webkit.WebChromeClient
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient

import com.toughra.ustadmobile.R
import com.ustadmobile.core.controller.ScormPackagePresenter
import com.ustadmobile.core.impl.UMAndroidUtil
import com.ustadmobile.core.view.ScormPackageView

class ScormPackageActivity : ZippedContentActivity(), ScormPackageView {

    private var mWebView: WebView? = null

    private var mPresenter: ScormPackagePresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scorm_package)

        mWebView = findViewById(R.id.activity_scorm_package_webview)
        if (Build.VERSION.SDK_INT >= 17) {
            mWebView!!.settings.mediaPlaybackRequiresUserGesture = false
        }

        mWebView!!.settings.javaScriptEnabled = true
        mWebView!!.settings.domStorageEnabled = true
        mWebView!!.settings.cacheMode = WebSettings.LOAD_DEFAULT
        mWebView!!.webViewClient = WebViewClient()
        mWebView!!.webChromeClient = WebChromeClient()


        setUMToolbar(R.id.um_toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        mPresenter = ScormPackagePresenter(this,
                UMAndroidUtil.bundleToMap(intent.extras), this)
    }

    override fun setTitle(title: String) {
        super.setTitle(title)
    }


    override fun onDestroy() {
        mPresenter!!.onDestroy()
        super.onDestroy()
    }

    override fun loadUrl(url: String) {
        mWebView!!.loadUrl(url)
    }
}
